frontend/packages/console-shared/src/components/formik-fields/SyncedEditorField.tsx

```diff
@@ -49,7 +64,27 @@ const SyncedEditorField: React.FC<SyncedEditorFieldProps> = ({
   const [sanitizeToCallback, setSanitizeToCallback] = React.useState<FormErrorCallback>(undefined);
   const [disabledFormAlert, setDisabledFormAlert] = React.useState<boolean>(formContext.isDisabled);
 
+  const isEditorTypeEnabled = (type: EditorType): boolean =>
+    !(type === EditorType.Form ? formContext?.isDisabled : yamlContext?.isDisabled);
+  const getEditorType = (): EditorType => {
+    if (
+      preferredEditorType &&
+      preferredEditorType !== PREFERRED_CREATE_EDIT_METHOD_USER_SETTING_VALUE_LATEST &&
+      isEditorTypeEnabled(preferredEditorType as EditorType)
+    ) {
+      return preferredEditorType as EditorType;
+    }
+    if (lastViewedEditorType && isEditorTypeEnabled(lastViewedEditorType)) {
+      return lastViewedEditorType;
+    }
+    return field.value as EditorType;
+  };
+
+  const [activeEditorType, setActiveEditorType] = React.useState<EditorType>(getEditorType);
```

Comment:
Why do you need another state for editor type? Wouldn't the formik state value work here?

Reply:
updated to custom component.